### DISCUSSION TITLE
Is it possible to have IC (via SolutionUserObject) for all the timesteps

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#26949] [Q&A General]
So normally we'd do the IC block like this (below), for the IC solution at T=0 

```
### ################################################################### #######
### INITIAL CONDITIONS
### ################################################################### #######

[ICs]
  [ic_porepressure_L1]
    type = FunctionIC
    variable = porepressure_L1
    function = fcn_pw_from_prev_results
  []
[]
[Functions]
  [fcn_pw_from_prev_results]
    type = SolutionFunction
    from_variable = porepressure_L1
    solution = uobj_prev_results
  []
[]
[UserObjects]
  [uobj_prev_results]
    type = SolutionUserObject
    mesh = 'OUTPUTExodus/XXXX2023_HR01_F001.e'
    # execute_on = 'INITIAL'
    # sytem_variables = porepressure_L1   ### THIS CAUSED AN ERROR: 'unused parameter'
    timestep = 'LATEST'
  []
[]
```

Now I'm wondering if we could have IC for all the timesteps (we have .e for all the timesteps), the reason being that the whole simulation take so long to converge (50+ NL steps), but we just want to adjust a bit of BC for the new run, so it doesn't have to go through long NL steps again

I suspect it gotta have something to do with [Controls] but still haven't found anything that might work.

Thank you!
Traiwit

**Comment by GiudGiud** [comment]
Hello

Use the IC for the first run

For every subsequent runs use a bunch of SolutionIC and a solutionUO loading the solution from the first run 

Guillaume

**Reply by Traiwit** [reply]
Hi @GiudGiud 

https://mooseframework.inl.gov/source/ics/SolutionIC.html

Thank you! so I need to have 1 `SolutionIC` and 1 `solutionUO`  per each time step and use [Controls] to control it?

**Reply by GiudGiud** [reply]
Umm actually no you can't do that. I thought you were using separate simulations. 

If you are using a single simulation, then the fields are not re initialized between each time step.
So where do you want this starting guess to come from?

**Reply by Traiwit** [reply]
>  I thought you were using separate simulations.

I also could do this but it would take 2 hrs just to set up (even before solving), the mesh has 18mil element :'(

yeh, that's what I wondering, if it possible, lets say I have all the .e solution from the previous run (I save each timestep result to each .e). but I want to adjust some BC (just a lil bit) for the next run

how can I use each .e from the previous run to be the initial guess for each timestep of the new run (at timestep_begin).

**Reply by GiudGiud** [reply]
I think what you want is a SolutionPredictor

It would use the solution user object every time step to set the initial guess for the nonlinear variables. 

Predictors are the system of choice for changing initial guesses in my understanding. We generally have only used them for the Predictor Corrector scheme but this seems like a fitting use.
